test/e2e/framework.go

```diff
@@ -548,6 +623,85 @@ func applyYAML(filename string, ns string) error {
 		return err
 	}
 	outStr, errStr := string(stdout.Bytes()), string(stderr.Bytes())
-	log.Printf("stdout: %s, stderr: %s", outStr, errStr)
+	if outStr != "" {
+		log.Printf("%s", outStr)
+	}
+	if errStr != "" {
+		log.Printf("%s", errStr)
+	}
+	return nil
+}
+
+func runCommand(cmd string) (string, error) {
+	err := wait.Poll(1*time.Second, defaultTimeout, func() (bool, error) {
+		var stdout, stderr bytes.Buffer
+		command := exec.Command("bash", "-c", cmd)
+		log.Printf("Running command %s", cmd)
+		command.Stdout = &stdout
+		command.Stderr = &stderr
+		err := command.Run()
+		if err != nil {
+			return false, nil
+		}
+		outStr, errStr := string(stdout.Bytes()), string(stderr.Bytes())
+		if outStr != "" {
+			log.Printf("%s", outStr)
+		}
+		if errStr != "" {
+			log.Printf("%s", errStr)
+			return false, nil
+		}
+		return true, nil
+	})
+	return "", err
+}
+
+func deleteYAML(filename string, ns string) error {
+	cmd := fmt.Sprintf("kubectl delete -f %s -n %s", filename, ns)
+	if ns == "" {
+		cmd = fmt.Sprintf("kubectl delete -f %s", filename)
+	}
+	var stdout, stderr bytes.Buffer
+	command := exec.Command("bash", "-c", cmd)
+	log.Printf("Deleing YAML file (%s)", cmd)
+	command.Stdout = &stdout
+	command.Stderr = &stderr
+	err := command.Run()
+	if err != nil {
+		return nil
+	}
+	outStr, errStr := string(stdout.Bytes()), string(stderr.Bytes())
+	if outStr != "" {
+		log.Printf("%s", outStr)
+	}
+	if errStr != "" {
+		log.Printf("%s", errStr)
+	}
 	return nil
 }
+
+func (data *TestData) waitForResourceExistOrNot(namespace string, resourceType string, resourceName string, shouldExist bool) error {
+	err := wait.Poll(1*time.Second, defaultTimeout, func() (bool, error) {
+		exist := true
+		tagScopeClusterKey := strings.Replace(common.TagScopeNamespace, "/", "\\/", -1)
+		tagScopeClusterValue := strings.Replace(namespace, ":", "\\:", -1)
+		tagParam := fmt.Sprintf("tags.scope:%s AND tags.tag:%s", tagScopeClusterKey, tagScopeClusterValue)
+		resourceParam := fmt.Sprintf("%s:%s AND display_name:*%s*", common.ResourceType, resourceType, resourceName)
+		queryParam := resourceParam + " AND " + tagParam
+		var cursor *string = nil
+		var pageSize int64 = 500
+		response, err := testData.nsxClient.QueryClient.List(queryParam, cursor, nil, &pageSize, nil, nil)
+		if err != nil {
+			return false, err
+		}
+		if len(response.Results) == 0 {
+			exist = false
+		}
+		log.Printf("QueryParam: %s Result: %t", queryParam, exist)
```

Comment:
this message will be output many time if resource status is not matched. output the result seems no value since from return value we could deduce the result.